src/sentry/migrations/0356_break_release_owner_foreign_key.py

```diff
@@ -0,0 +1,53 @@
+# Generated by Django 2.2.28 on 2023-02-12 02:31
+
+from django.db import migrations
+
+import sentry.db.models.fields.hybrid_cloud_foreign_key
+from sentry.new_migrations.migrations import CheckedMigration
+
+
+class Migration(CheckedMigration):
+    # This flag is used to mark that a migration shouldn't be automatically run in production. For
+    # the most part, this should only be used for operations where it's safe to run the migration
+    # after your code has deployed. So this should not be used for most operations that alter the
+    # schema of a table.
+    # Here are some things that make sense to mark as dangerous:
+    # - Large data migrations. Typically we want these to be run manually by ops so that they can
+    #   be monitored and not block the deploy for a long period of time while they run.
+    # - Adding indexes to large tables. Since this can take a long time, we'd generally prefer to
+    #   have ops run this and not block the deploy. Note that while adding an index is a schema
+    #   change, it's completely safe to run the operation after the code has deployed.
+    is_dangerous = False
+
+    dependencies = [
+        ("sentry", "0355_break_groupowner_user_foreign_key"),
+    ]
+
+    database_operations = [
+        migrations.AlterField(
+            model_name="release",
+            name="owner",
+            field=sentry.db.models.fields.foreignkey.FlexibleForeignKey(
+                to="sentry.User", db_constraint=False, db_index=True, null=True, blank=True
+            ),
+        ),
+    ]
+
+    state_operations = [
+        migrations.AlterField(
+            model_name="release",
+            name="owner",
+            field=sentry.db.models.fields.hybrid_cloud_foreign_key.HybridCloudForeignKey(
+                "sentry.User", blank=True, db_index=True, null=True, on_delete="SET_NULL"
+            ),
+        ),
+        migrations.RenameField(
+            model_name="release",
+            old_name="owner",
+            new_name="owner_id",
+        ),
+    ]
+
+    operations = database_operations + [
+        migrations.SeparateDatabaseAndState(state_operations=state_operations)
+    ]
```

Comment:
Should the `database_operations` be inside of `SeparateDatabaseAndState()`? Right now the `database_operations` will apply DDL and state changes.

Reply:
Replied elswhere, but applying the database operations as state changes, then the state changes, does work.  It seems to "fix" an issue with `migratesql` that does not generate the SQL correctly otherwise :/